src/main.rs

```diff
@@ -1202,115 +1203,113 @@ fn cleanup_autogenerated_file(path: &Path) {
-//
-//     return success
+fn parse_manpage_at_path(
+    manpage_path: &Path,
+    output_directory: &Path,
+    deroff_only: bool,
+    write_to_stdout: bool,
+) -> io::Result<bool> {
+    // Clear diagnostic
+    // diagnostic_output[:] = []
+    // diagnostic_indent = 0
+
+    // Set up some diagnostic
+    // add_diagnostic(format!("Considering {}", manpage_path));
+    // diagnostic_indent += 1
+
+    let mut manpage = String::new();
+    if manpage_path.ends_with(".gz") {
+        let mut gz = GzDecoder::new(File::open(manpage_path)?);
+        gz.read_to_string(&mut manpage)?;
+    } else if manpage_path.ends_with(".bz2") {
+        let mut bz = BzDecoder::new(File::open(manpage_path)?);
+        bz.read_to_string(&mut manpage)?;
+    } else if manpage_path.ends_with(".xz") || manpage_path.ends_with(".lzma") {
+        let mut xz = XzDecoder::new(File::open(manpage_path)?);
+        xz.read_to_string(&mut manpage)?;
+    } else if [".1", ".2", ".3", ".4", ".5", ".6", ".7", ".8", ".9"]
+        .iter()
+        .any(|suffix| manpage_path.ends_with(suffix))
+    {
+        File::open(manpage_path)?.read_to_string(&mut manpage)?;
+    }
+
+    // Get the "base" command, e.g. gcc.1.gz -> gcc
+    // These casts are safe as OsStr is internally a wrapper around [u8] on all
+    // platforms. Taken from libstd.
+    let cmdname = manpage_path
+        .file_name()
+        .and_then(|file| file.as_bytes().splitn(2, |b| *b == b'.').next());
+    let cmdname = String::from_utf8_lossy(cmdname.unwrap());
+    let ignored_commands = [
+        "cc", "g++", "gcc", "c++", "cpp", "emacs", "gprof", "wget", "ld", "awk",
+    ];
+    if ignored_commands.contains(&cmdname.as_ref()) {
+        return Ok(false);
+    }
+
+    // Ignore perl's gazillion man pages
+    let ignored_prefixes = ["perl", "zsh"];
+    if ignored_prefixes
+        .iter()
+        .any(|prefix| cmdname.starts_with(prefix))
+    {
+        return Ok(false);
+    }
+
+    // Ignore the millions of links to BUILTIN(1)
+    if manpage.contains("BUILTIN 1") || manpage.contains("builtin.1") {
+        return Ok(false);
+    }
+
+    let parsers = if deroff_only {
+        &[ManType::TypeDeroff(TypeDeroff)]
+    } else {
+        ManType::ALL
+    };
+    let parsers = parsers.iter().filter(|parser| parser.is_my_type(&manpage));
+    let mut parsers = parsers.peekable();
+
+    if parsers.peek().is_none() {
+        // add_diagnostic(format!("{}: Not supported", manpage_path));
+    }
+
+    if let Some(mut completions) = parsers
+        // .inspect(|parser| add_diagnostic(format!("Trying {}", parser)))
+        .find_map(|parser| parser.parse_man_page(&manpage, &cmdname))
+    {
+        let comments = format!(
+            "# {}\n#Autogenerated from man page {}\n",
+            &cmdname,
+            manpage_path.display()
+        );
+        completions.insert_str(0, &comments);
+
+        if write_to_stdout {
+            let stdout = io::stdout();
```

Comment:
```rust
io::stdout()
    .lock()
    .write_all(completions.as_bytes())?;
```

Reply:
Nice, looks like a good option. Adding this.